### ISSUE TITLE
Choosing values in "Pyramids" example

### ISSUE THREAD

**Issue by phatal** [Unity-Technologies/ml-agents#902]
Hi, can you plz explain how this values were choosen?

`public override void CollectObservations()`
`{`
`        if (useVectorObs)`
`        {`
`            const float rayDistance = 35f;`
`            float[] rayAngles = {20f, 90f, 160f, 45f, 135f, 70f, 110f};`
`            float[] rayAngles1 = {25f, 95f, 165f, 50f, 140f, 75f, 115f};`
`            float[] rayAngles2 = {15f, 85f, 155f, 40f, 130f, 65f, 105f};`
`            string[] detectableObjects = {"block", "wall", "goal", "switchOff", "switchOn", "stone"};`
`            AddVectorObs(rayPer.Perceive(rayDistance, rayAngles, detectableObjects, 0f, 0f));`
`            AddVectorObs(rayPer.Perceive(rayDistance, rayAngles1, detectableObjects, 0f, 5f));`
`            AddVectorObs(rayPer.Perceive(rayDistance, rayAngles2, detectableObjects, 0f, 10f));`
`            AddVectorObs(switchLogic.GetState());`
`            AddVectorObs(transform.InverseTransformDirection(agentRb.velocity));`
`        }`
`    }`

More precisely, I mean Why there 3 rayAngles arrays and how values for each of it were choosen?

I'm asking cuz I'm trying to use RayPerception at my project and wanna to collect best practices.

**Comment by awjuliani** [comment]
Hi @phatal,

While it looks somewhat daunting, the set-up here is actually relatively straightforward. Three sets of rays are used for three different heights of vision. One straight ahead, one upward 5 meters, and the third upward 10 meters. That is what the final argument of the `rayPer.Perceive` does. Each `rayAngle` list describes the angle from the right side of the agent to send the rays. Each corresponds to a spread from the forward direction of the agent. The reason they are all slights off from each other is to allow the different sets of rays to not only be at different heights, but also to possibly pick up on things the others rays might miss. I hope that is helpful. 

If you want to visualize what is happening, go into the editor and start play mode. Then look in the scene view with the `Gizmos` button enabled. You will see black lines corresponding to each of the rays.

**Comment by phatal** [comment]
Thank you very much, now it become clear. I know that by rules I must use 1 issue for 1 question, and if it is vialotion I will create new issue then, but now I wanna ask one more question here:
I wanna to make simple clone of soccer stars game with where team will be controled by ML agents (just for learning myself). For what would be easier to explain I will attach screenshot:

![image](https://user-images.githubusercontent.com/14150045/42123779-e4a36a02-7c81-11e8-993f-97746883d890.png)

1. 2 teams. 5 units (footballers) at each 1 ball 2 goals. units in team do not divide into a goalkeeper and usual player (all the same)

2. Turn based game. Players make moves one by one.

3. During turn player choose which unit will kick, and then drag him to set him power and direction in 2d space (dragging is just anim and not neccesary for ML agent players of course, therefore just set power and direction) and then the given force is applied to the unit.

4. Unit can collide with other units, ball, walls, goal.

5. Turn is considered complete when all moving units stop or when ball hits the goal. if at the end of the turn any unit is inside goal (except ball of course), then it will be moved to any free position at some distance from the goal.

6. Then the turn goes to another player.

7. Win team which will score first 3 goals (for example 3. can the number be different).

Summarize this game such a mix of footbal and airhockey with turn based gameplay.

When I was thinking how to solve this problem, I first of all thought that it would be logical to train agent from simple to complex and decide that first task will be to learn player score the goal by collecting dist from player to main parts of env like ball, walls, goal (top and bottom point to know size of goal) and appliyng power and direction to unit (changing ```localEulerAngles``` and,```addforce``` to ```RigidBody``` along red axis). I spent about 2-3 days triyng to realize that result wasn't satysfying. Then I decide to make a task more simple and train agent just to kick a ball. So...it was like that I coped with this task (default ppo), but then I become stuck with problem how to use my already trained model to solve next task about scoring the goal, and even more how to use later that model with task when at field will be all 10 players (5 each team). And at that moment I realized that I need help. Of course I'm not asking you to explain me solution in every small details, but any info will be useful for me. 
Thank you in advance and I apologize for the level of my English and using googltranslate sometimes for translating text )))
